compiler/src/iree/compiler/Codegen/Common/MaterializeTuningSpecsPass.cpp

```diff
@@ -107,6 +107,16 @@ getUserTuningSpec(ModuleOp module, IREE::Codegen::IREECodegenDialect &dialect) {
            << clCodegenTuningSpecPath;
   }
 
+  // Iterate through all operations in the module to verify attributes
+  for (Operation &op : (*maybeTransformLibrary).getBody()->getOperations()) {
+    for (NamedAttribute attr : op.getAttrs()) {
+      if (failed(dialect.verifyOperationAttribute(&op, attr))) {
+        return op.emitError() << "Attribute verification failed for operation "
+                                 "in the user tuning spec";
+      }
+    }
+  }
```

Comment:
We should call verify on the whole module. You can see this used here: https://github.com/llvm/llvm-project/blob/57c161a6479fb70a31553e2f9bc1efa46262aa92/mlir/lib/Dialect/Transform/Transforms/TransformInterpreterUtils.cpp#L118

Reply:
Actually, I don't think this is needed along this code path because the user spec verification happens in `parseTransformModuleFromFile`. We should verify linked specs after linking though.l